accepted/2020/platform-exclusion/platform-exclusion.md

```diff
@@ -0,0 +1,217 @@
+support the called API yet.
+
+This feature made the following assumption:
+
+* An unmarked API is considered to work on all OS platforms.
+* An API marked with `[MinimumOSPlatform("...")]` is considered only portable to
+  the specified OS platforms (please note that the attribute can be applied
+  multiple times).
+
+This approach works well for cases where the API is really calling an
+OS-specific API. Good examples include the Windows Registry or Apple-specific
+graphic APIs. This is very different from browser support for specific HTML and
+CSS features: those generally aren't browser specific, which is why browser
+checks never worked well for the web. But for OS specific APIs, asking whether
+you're actually running on that OS is very workable. The only fragile aspect
+here is getting the version number right, but the analyzer enforces that.
+
+However, this approach doesn't work well for APIs that represent concepts that
+aren't OS specific, such as threading or I/O. This is a problem for OS platforms
+that are more constrained, such as Blazor Web Assembly, which runs in a browser
+sandbox and thus can't freely create threads or open random files on disk.
+
+One could, in principle, apply the `[MinimumOSPlatform("...")]` attribute for
+all other OS platforms that do support them, but this has several downsides:
+
+1. The so marked APIs are now considered platform-specific, which means that
+   just calling the API from portable code will now always be flagged by the
+   analyzer.
+2. Every time .NET adds another operating system all call sides guarding calls
+   to these APIs now also have to check for this new OS.
+
+In this spec, we're proposing a mechanism by which we can mark APIs as
+unsupported.
+
+## Scenarios and User Experience
+
+* Building a Blazor Web Assembly app
+    - Using an API that is unsupported will be flagged by the analyzer.
+* Building a class library
+    - Building a regular class library (targeting `netstandard` or `net5.0`)
+      will not complain about APIs that are unsupported in Blazor.
+* Building a class library for use in Blazor Web Assembly
+    - Developer can modify the project file to explicitly indicate that they
+      want to see issues with APIs that are unsupported in Blazor.
+
+## Requirements
+
+### Goals
+
+* Be able to mark APIs that are unsupported on a particular OS
+* Allows platforms to be considered supported by default with the ability to add
+  or remove platforms from the project file and SDK.
+* Blazor Web Assembly is only considered supported when building a Blazor Web
+  Assembly app. For regular class libraries (`netstandard` or `net5.0`) the
+  default assumes that it's not supported and thus won't generate any warnings.
+  However, the developer must still be able to manually include Blazor Web
+  Assembly manually.
+* The annotations for unsupported APIs can express that an API was only
+  unsupported for a specific version.
```

Comment:
A `specific version`? or a range of versions?

Reply:
Range, I'll reword that